pkg/provider/kubernetes/client.go

```diff
@@ -107,10 +108,7 @@ func (c *Client) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushRemot
 	return c.fullDelete(ctx, remoteRef.GetRemoteKey())
 }
 
-func (c *Client) PushSecret(ctx context.Context, value []byte, typed v1.SecretType, _ *apiextensionsv1.JSON, remoteRef esv1beta1.PushRemoteRef) error {
-	if remoteRef.GetProperty() == "" {
-		return fmt.Errorf("requires property in RemoteRef to push secret value")
-	}
+func (c *Client) PushSecret(ctx context.Context, values map[string][]byte, typed v1.SecretType, _ *apiextensionsv1.JSON, remoteRef esv1beta1.PushRemoteRef) error {
```

Comment:
@moolen So this is how I'd imagine this would work. I'll have to implement this for all providers, but some of them won't have a remote property. I'll examine that case by case. But since it will only push a single thing I guess I will have to somehow get that single value ( like a for loop ) and push that to maintain the backwards compatibility.

Is this the right approach you were looking for?

Reply:
#2859 looks really promising and much better than what i've had in mind in terms of signature change :broom: :sweat_smile: 
But we still need to change the value(s) type [as you said](https://github.com/external-secrets/external-secrets/pull/2859#issuecomment-1797845435), what you suggest looks good to me :+1:

Reply:
Nice! Excellent! Thanks very much. I'm gonna crack on with it then. 😊😊😊